src/lib/components/forms/EditProject.svelte

```diff
@@ -0,0 +1,75 @@
+<!-- @component
+Edit project metadata
+-->
+<script lang="ts">
+  import type { Project } from "$lib/api/types";
+
+  import { enhance } from "$app/forms";
+
+  import { createEventDispatcher } from "svelte";
+  import { _ } from "svelte-i18n";
+
+  import Button from "../common/Button.svelte";
+  import Field from "../inputs/Field.svelte";
+  import Flex from "../common/Flex.svelte";
+  import Switch from "../inputs/Switch.svelte";
+  import Text from "../inputs/Text.svelte";
+  import TextArea from "../inputs/TextArea.svelte";
+
+  import { pinned } from "$lib/components/projects/ProjectPin.svelte";
+
+  import { canonicalUrl } from "$lib/api/projects";
+
+  export let project: Partial<Project> = {};
+
+  const dispatch = createEventDispatcher();
+
+  $: action = project?.id
+    ? new URL("?/edit", canonicalUrl(project as Project)).href
+    : "/documents/projects/";
+
+  // handle optimistic updates
+  $: is_pinned = $pinned.includes(project as Project) || project.pinned;
+
+  function onSubmit({ submitter }) {
+    submitter.disabled = true;
+    return ({ result, update }) => {
+      update(result);
+      dispatch("close");
+    };
+  }
+</script>
+
+<form {action} method="post" use:enhance={onSubmit}>
+  <Flex direction="column" gap={1}>
+    <Field title={$_("projects.fields.title")} required>
+      <Text
+        name="title"
+        placeholder={$_("projects.fields.title")}
+        value={project.title}
+      />
+    </Field>
+    <Field title={$_("projects.fields.description")}>
+      <TextArea name="description" value={project.description} />
+    </Field>
+    <Field title={$_("projects.fields.private")} inline>
+      <Switch name="private" checked={project.private} />
+    </Field>
+    <Field title={$_("projects.fields.pinned")} inline>
+      <Switch name="pinned" checked={is_pinned} />
+    </Field>
+
+    <Flex class="buttons">
+      <Button type="submit" mode="primary" full>{$_("edit.save")}</Button>
```

Comment:
If this form can be used to both edit an existing or create a new project, we should allow the submit button's label to be configurable via prop.